### ISSUE TITLE
Using regression layer with Euclidean Loss to predict count from an image, but the probabilities are same in inference?

### ISSUE THREAD

**Issue by h612** [vlfeat/matconvnet#977]
I'm using AlexNet for training a classification of count.( MatConvNet, DagNN). I've 35 classes of counts. I'm trying use euclidean loss regression. Here is the network end layers

```
net.addLayer('fc7', dagnn.Conv('size', [1 1 4096 4096], 'hasBias', true, 'stride', [1, 1], 'pad', [0 0 0 0]), {'bn6'}, {'fc7'},  {'conv7f'  'conv7b'});
net.addLayer('relu7', dagnn.ReLU(), {'fc7'}, {'relu7'}, {});
net.addLayer('bn7',dagnn.BatchNorm('numChannels',4096),{'relu7'},{'bn7'},{'bn7f','bn7b','bn7m'});

classLabels=max(unique(imdb_32.images.labels));
net.addLayer('classifier', dagnn.Conv('size', [1 1 4096 35], 'hasBias', true, 'stride', [1, 1], 'pad', [0 0 0 0]), {'bn7'}, {'classifier'},  {'conv8f'  'conv8b'});
net.addLayer('prediction', dagnn.SoftMax(), {'classifier'}, {'prediction'}, {});

net.addLayer('objective', dagnn.Loss('loss', 'nnl2'), {'prediction', 'label'}, {'objective'}, {});
net.addLayer('error', dagnn.Loss('loss', 'classerror'), {'prediction','label'}, 'error') ;
```

The error after training is 0.55 and obj is 3.3. When I try inference, net.eval({'input', im_});

```
% obtain the CNN otuput
scores = net.vars(net.getVarIndex('prediction')).value;
scores = squeeze(gather(scores))

% show the classification results
[bestScore, best] = max(scores);
```

The scores are all equal to 0.0264. And it picks the first class- which has a count 16, and outputs that as the estimate count of patch. Which is wrong. Kindly help!?
